contracts/AirDrop.sol

```diff
@@ -0,0 +1,242 @@
+ * members of crypto-project
+ */
+contract AirDrop is EIP712, IAirDrop {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @dev Store balances of members
+     * @return amounts of available tokens & ether
+     */
+    mapping(address => uint256) public balanceOfTokens;
+    mapping(address => uint256) public balanceOfEther;
+
+    /**
+     * @dev Use for sign message with EIP712 standart
+     * @notice 'DROP_HASH' is signature of primaryType 'Drop'
+     */
+    bytes32 public constant DROP_HASH =
+        keccak256("Drop(uint256 amount,uint256 deadline,address recepient)");
+
+    address private _owner;
+
+    IERC20 public token;
+
+    /**
+     * @dev modifier which contains conditions who's can call functions
+     *
+     * NOTE : if function have 'onlyOwner' thats mean call this function
+     * can only address which contained in '_owner'
+     */
+    modifier onlyOwner() {
+        require(_owner == msg.sender, "Error : caller is not the owner");
+        _;
+    }
+
+    /**
+     * @dev Set 'token' IERC20 to interact with thrid party token
+     *
+     * Put address to state variable '_owner'
+     *
+     * @param token_ - of ERC20 contract
+     */
+    constructor(address token_) EIP712("AirDrop", "1") {
+        require(
+            token_.code.length > 0,
+            "Error : Incorrect address , only contract address"
+        );
+        token = IERC20(token_);
+        _owner = msg.sender;
+    }
+
+    /**
+     * @dev send ERC20 tokens from 'owner_' balance to SC
+     *
+     * @param amount_ - how many tokens deposits
+     *
+     * NOTE : this function use wrapper of safeERC20 library
+     * check 'SafeERC20.sol' for more information
+     */
+    function depositTokens(uint256 amount_) external override onlyOwner {
+        require(amount_ != 0, "Error : 'Amount' , equal to 0");
+        token.safeTransferFrom(msg.sender, address(this), amount_);
+        emit DepositTokens(amount_);
+    }
+
+    /**
+     * @dev send ethers from 'owner_' balance to SC
+     *
+     * NOTE : if you send ether to 0 address , ethers will be blocked.
+     */
+    function depositEther() external payable override onlyOwner {
+        require(msg.value != 0, "Error : 'Amount' , equal to 0");
+        uint256 value = uint256(msg.value);
+        emit DepositEther(value);
+    }
+
+    /**
+     * @dev make tokens available for members to withdraw
+     *
+     * @param recepients_ - address of member
+     * @param amount_ - how many tokens will be available
+     * @param deadline_ - how long signature will be valid
+     * @param v_ - can be only 27 or 28
+     * @param r_ - contain info to recover signer
+     * @param s_ - contain info to recover signer
+     *
+     * NOTE : more information 'EIP712'
+     */
+    function dropTokens(
+        address recepients_,
+        uint256 amount_,
+        uint256 deadline_,
+        uint8 v_,
+        bytes32 r_,
+        bytes32 s_
+    ) external override onlyOwner {
```

Comment:
You have a Signature Replay security issue

Reply:
first: owner can use the same signature many times (read about nonce in signature) 
second: owner can use the same signature for both dropTokens and dropEthers function. let's make it impossible. Advise: add flag "bool isEther " to Drop container as example